### PR TITLE
feat: migrate dialog to radix

### LINKED ISSUE #92: migrate dialog from headless to radix

### PR DESCRIPTION
**Pull request by snymnd**
# Description & Technical Solution
- [feat: migrate dialog to radix](https://github.com/ppdbultimate/design-system-moon/pull/93/commits/209ee5ac237c3c12e50559e4f76fcb6df641be1e)
- [feat: adjust modal and drawer z-index](https://github.com/ppdbultimate/design-system-moon/pull/93/commits/ba4886b17ac0bba032802a7252fdae73bc573ee3)

# Checklist

- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] Already rebased against the main branch.

# Deployment Links To Test
- https://design-system-moon-git-i92-migratedialogfro-e93978-ppdbultimate.vercel.app/sandbox/dialog
- https://design-system-moon-git-i92-migratedialogfro-e93978-ppdbultimate.vercel.app/sandbox/adaptive-modal

# Screenshots
image
image

[dialog.webm](https://github.com/ppdbultimate/design-system-moon/assets/70748569/86124f80-004e-4140-83ff-1be240fb46db)

# Related issue

- Resolve #92

### CODE REVIEW

**Comment by ppdbultimate** [comment]
#### 📝 Changed routes:


| [**…s/ExampleAdaptiveModal**](https://design-system-moon-p272glp0d-ppdbultimate.vercel.app/sandbox/components/ExampleAdaptiveModal) |
|:-:|
|<a href="https://design-system-moon-p272glp0d-ppdbultimate.vercel.app/sandbox/components/ExampleAdaptiveModal">Screenshot of /sandbox/components/ExampleAdaptiveModal</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/1r_U-ttIgb-0k_ZMGVbMqoiX6QW0RfNRZxDnzbPUWLJRtlT3x5ZtHaQ0DvCz9p-cOTek-5b3K0moKleezwiung.png">(view full size)</a> |







Commit 1fe7479e0b4c2d82e4ddd71cdcda9afbc6459904 (https://design-system-moon-p272glp0d-ppdbultimate.vercel.app).

**Review by rizqitsani — APPROVED** [review]

File: src/components/dialog/AlertDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+const AlertDialogFooter = ({
+  className,
+  ...props
+}: React.HTMLAttributes<HTMLDivElement>) => (
+  <div
+    className={clsxm(
+      'border-t border-typo-divider rounded-b-2xl',
+      '-mb-4 sm:-mb-6 -mx-4 sm:-mx-6',
+      'px-4 sm:px-6 py-4',
```
Comment (winatungmiharja) [inline src/components/dialog/AlertDialog.tsx:73]:
```suggestion
      'px-4 sm:px-6 py-4 sm:py-6',
```
ini kelupaan nus
Screenshot 2024-01-26 004328

**Review by winatungmiharja — CHANGES_REQUESTED** [review]

File: src/components/dialog/AlertDialog.tsx (hunk 1 above)
Comment (rizqitsani) [reply src/components/dialog/AlertDialog.tsx:73]: Jadi p-4 sm:p-6 aja ya wkwkwkwk

Comment (snymnd) [reply src/components/dialog/AlertDialog.tsx]: done

**Review by winatungmiharja — APPROVED** [review]

**Pull request merged**